sdk/src/main/java/io/radar/sdk/RadarSettings.kt

```diff
@@ -94,9 +98,32 @@ internal object RadarSettings {
     }
 
     internal fun getTrackingOptions(context: Context): RadarTrackingOptions {
-        val optionsJson = getSharedPreferences(context).getString(KEY_TRACKING_OPTIONS, null) ?: return RadarTrackingOptions.EFFICIENT
-        val optionsObj = JSONObject(optionsJson)
-        return RadarTrackingOptions.fromJson(optionsObj)
+        val optionsJson = getSharedPreferences(context).getString(KEY_TRACKING_OPTIONS, null)
+        val options: RadarTrackingOptions?
+        if (optionsJson != null) { // v3 tracking options set
+            val optionsObj = JSONObject(optionsJson)
+            options = RadarTrackingOptions.fromJson(optionsObj)
+        } else {
```

Comment:
I'm a bit confused on what these changes are needed for - is this basically to migrate old v2 tracking options to v3?

Reply:
Whereas v3 has custom tracking options, v2 only had priority (responsiveness or efficiency), sync (all or possible state changes), and offline replay (stops or none). These changes ensure that if a customer upgrades from SDK v2 to SDK v3, devices are migrated to the corresponding new tracking options on upgrade. Without these changes, a customer on responsiveness mode in v2 will be on the efficient preset in v3 until the `startTracking()` codepath is hit.

Reply:
awesome, great, that's along the lines of what I figured - mind capturing that in a comment above this block?